python/iwp/data_loader.py

```diff
@@ -513,3 +515,64 @@ def add_timestep_as_coord( ds ):
                             preprocess=add_timestep_as_coord )
 
     return ds
+
+
+    def get_data_loader( dataset_path_pattern,
+                         time_indices,
+                         variables=[],
+                         xy_slice_indices=[],
+                         batch_size=1000,
+                         shuffle=True,
+                         num_workers=1 ):
+        """
+        Initializes an DataLoader object that has been seeded with an 
+        IWPDataset. The DataLoader acts as a pointer that can be iterated
+        through to get BATCH_SIZE xy slices.
+
+        Takes 7 arguments:
+
+          dataset_path_pattern - Path pattern of the dataset, with a format specifier to instantiate
+                                 into a path for each time step.  Each time step's dataset path is
+                                 created by formatting dataset_path_pattern with the corresponding
+                                 index in time_indices.
+          time_indices         - Sequence of time step indices to create the dataset from.  Scalar
+                                 integers may also be specified to indicate a specific time step.
+          variables            - Optional list of variable names in the netCDF files to extract XY
+                                 slices when creating the dataset.  If omitted, defaults to all of
+                                 the grid variables in te netCDF4 dataset's first time step file
+                                 (i.e. all 3D, floating point variables in the file associated
+                                 with time_indices[0])).
+
+                                 NOTE: The order of the concatenated XY slices matches the order of
+                                       variables.
+
+          xy_slice_indices     - Optional sequence of XY slice indices to create the dataset from.  If
+                                 omitted, defaults to all XY slices present.  Scalar integers may
+                                 also be specified to indicate a specific XY slice.
+          batch_size           - Optional integer number of xy slices to be returned per batch.
+                                 Defaults to a batch size of 1000.
+          shuffle              - Optional boolean denoting whether the data should be reshuffled
+                                 after every epoch has been processed. Defaults to True.
+          num_workers          - Optional integer number of subprocesses to use for data loading.
```

Comment:
The default specified here is inconsistent with the actual default.  What does num_workers=1 do?  Spawn one background process independent of the main?

Style: Use number_workers instead of num_workers.

Reply:
I'm honestly not 100% certain but it seems like that is the case. 

https://discuss.pytorch.org/t/guidelines-for-assigning-num-workers-to-dataloader/813/2